### DISCUSSION TITLE
AR mode shows object too small

### DISCUSSION THREAD

**Discussion by sercandemirtas** [google/model-viewer#1906] [Q&A]
Hello all!

![IMG_0416](https://user-images.githubusercontent.com/68244546/88289269-ac6f7800-ccfd-11ea-897b-8ebc678c1b64.PNG)

 Object is too small on AR. Anyone knows whats the problem ? Ps: I designed obj via Blender. and tried it with scaling object maximum but still when I open in AR mode it goes too small. As you see in the picture its %1000 zoom  but still its too small. 

Thanks

**Comment by elalish** [comment]
Cute jacket! The trouble is that glTF specifies that the units for all models are meters. Most modeling programs are pretty loose with their units, so it's easy to accidentally export something that is at completely the wrong scale. It looks like this one might be in units of km? A top-level transform of 100x or 1000x will probably fix it. 

If that doesn't work, you might want to check some other things. Sometimes models are placed at a position that is absurdly far from the origin, and that can confuse things down the road. You might also want to check if you get the same results in our WebXR, SceneViewer and QuickLook AR modes.

**Comment by sercandemirtas** [comment]
> Cute jacket! The trouble is that glTF specifies that the units for all models are meters. Most modeling programs are pretty loose with their units, so it's easy to accidentally export something that is at completely the wrong scale. It looks like this one might be in units of km? A top-level transform of 100x or 1000x will probably fix it.
> 
> If that doesn't work, you might want to check some other things. Sometimes models are placed at a position that is absurdly far from the origin, and that can confuse things down the road. You might also want to check if you get the same results in our WebXR, SceneViewer and QuickLook AR modes.


Thank you dude. Fixed..

**Comment by sercandemirtas** [comment]
I have one more question to you. What about lights on AR mode ? As you see below Web version and AR version are different. Can we make light setup for AR mode ?


![IMG_0424](https://user-images.githubusercontent.com/68244546/88315673-44ca2480-cd1f-11ea-9281-94eec728237f.jpg)

**Comment by elalish** [comment]
Great! The lighting is intentionally different in AR, as both SceneViewer and QuickLook attempt to estimate the local lighting in the room and apply that to make the object seem more real (but it's ML, so it's not always perfect). WebXR doesn't have this feature yet, so right now the only way to get consistent lighting between 3D and AR in model-viewer is to use the default WebXR AR mode (only available on Chrome Android). See #964

**Comment by Saicopate** [comment]
What is the maximum size of the object displayed in AR? 
I found that some large objects (etc building 3D models) are auto-resized to fit a specific bounding box.

**Comment by elalish** [comment]
@Saicopate I believe that is a feature of SceneViewer (and possibly QuickLook?). WebXR mode does not currently auto scale at all. I don't actually know the answer, but hopefully their documentation will say something about it.

**Comment by polishchuk37a** [comment]
Hello! I have a project with model-viewer library and i need to set width and height for 3D model to show it in AR with setted size(width and height). Can i make this and how? 

I have been searching for the answer for three days and have not find yet. Help me please.

**Reply by elalish** [reply]
https://modelviewer.dev/docs/index.html#entrydocs-scenegraph-attributes-scale

**Reply by polishchuk37a** [reply]
Do you know what units of measurement does the scale attribute use? I mean for X, Y, Z

**Reply by elalish** [reply]
Scale is unitless - it's a multiplier. 1 is no change, 2 is 2x. glTF specifies that all files have units of meters, so if you're object is a strange size, it's probably a unit error on export.

**Reply by polishchuk37a** [reply]
Hello again! Can you give me answer for one question? 

I want to add button "Buy now" in opened AR mode. Have i possibillity to make this and how?
It must looks like this (look at picture):
![msg774986687-1050414](https://github.com/google/model-viewer/assets/55455148/86b7ec72-1f73-4ec6-b621-ffafda66d71b)

**Reply by elalish** [reply]
In WebXR mode, just add child DOM elements - follow this example: https://modelviewer.dev/examples/augmentedreality/#webXR

For Quick Look or Scene Viewer, their API is controlled by hash and query params on the URL, which you can append to `src` and we'll pass them along for you. They're pretty limited in what they can do though. See https://modelviewer.dev/docs/#entrydocs-augmentedreality-events-quickLookButtonTapped

**Reply by polishchuk37a** [reply]
Thank you so much :)

**Reply by polishchuk37a** [reply]
Using your recommendations above i can place button only in preview, not in opened AR mode.
I need to place button exactly in opened AR mode. So i didn't find solution to make this. Do you have any another variant to place button in opened AR mode?

**Reply by elalish** [reply]
You need to specifically scope your CSS with `ar-status`, as in the example above: https://modelviewer.dev/docs/#entrydocs-augmentedreality-css-arStatus

**Reply by polishchuk37a** [reply]
Hello! Can I place a 3d object with a given distance from the wall to the 3d object?

**Reply by elalish** [reply]
Hmm, we do placement by bounding box, so your best bet is to add some tiny, invisible geometry at the wall anchor point. We recommend the same technique to automatically frame animated models.

**Reply by polishchuk37a** [reply]
Thank you for your answer :)
